src/twin/system_info.rs

```diff
@@ -118,4 +204,112 @@ impl SystemInfo {
         .await
         .context("report: send")
     }
+
+    async fn metrics_handler(&mut self) -> Result<()> {
+        let time = match time::OffsetDateTime::now_utc().format(&Rfc3339) {
+            Ok(time) => time,
+            Err(e) => {
+                warn!("timestamp could not be generated: {e}");
+                return Ok(());
+            }
+        };
+
+        self.sysinfo_components.refresh_list();
+        self.sysinfo_memory.refresh_cpu_usage();
+        self.sysinfo_memory.refresh_memory();
+        self.sysinfo_disk.refresh();
+
+        let Some(hostname) = System::host_name() else {
+            warn!("hostname could not be read");
+            return Ok(());
+        };
```

Comment:
@mlilien @JoergZeidler imho better fail but  not sure.

Reply:
fail

Reply:
changed to:
        let Some(hostname) = System::host_name() else {
            bail!("metrics: hostname could not be read")
        };

Reply:
hmm, only error! log and return Ok(()), or bail! to restart the ods?

Reply:
@mlilien I meant to `bail!`

Reply:
sry, i meant that. as long as we don't panic, i',m fine.

Reply:
done